src/custom/pages/Profile/LockedGnoVesting/hooks.ts

```diff
@@ -0,0 +1,129 @@
+import TOKEN_DISTRO_ABI from 'abis/TokenDistro.json'
+import { MerkleDrop, TokenDistro } from 'abis/types'
+import { useSingleCallResult } from 'lib/hooks/multicall'
+import { useTransactionAdder } from 'state/enhancedTransactions/hooks'
+import { useContract } from 'hooks/useContract'
+import { COW as COW_TOKENS } from 'constants/tokens'
+import { SupportedChainId } from 'constants/chains'
+import { OperationType } from 'components/TransactionConfirmationModal'
+import { fetchClaim } from './claimData'
+
+// We just generally use the mainnet version. We don't read from the contract anyways so the address doesn't matter
+const COW = COW_TOKENS[SupportedChainId.MAINNET]
+
+export const MERKLE_DROP_CONTRACT_ADDRESSES: Record<number, string> = {
+  [SupportedChainId.MAINNET]: '0x64646f112FfD6F1B7533359CFaAF7998F23C8c40',
+  [SupportedChainId.RINKEBY]: '0x5444c4AFb2ec7f7367C10F7732b8558650c5899F',
+  [SupportedChainId.XDAI]: '0x3d610e917130f9D036e85A030596807f57e11093',
+}
+
+const useMerkleDropContract = () => useContract<MerkleDrop>(MERKLE_DROP_CONTRACT_ADDRESSES, MERKLE_DROP_ABI, true)
+
+export const TOKEN_DISTRO_CONTRACT_ADDRESSES: Record<number, string> = {
+  [SupportedChainId.MAINNET]: '0x68FFAaC7A431f276fe73604C127Bd78E49070c92',
+  [SupportedChainId.RINKEBY]: '0xeBA8CE5b23c054f1511F8fF5114d848329B8258d',
+  [SupportedChainId.XDAI]: '0x3d610e917130f9D036e85A030596807f57e11093',
+}
+
+const useTokenDistroContract = () => useContract<TokenDistro>(TOKEN_DISTRO_CONTRACT_ADDRESSES, TOKEN_DISTRO_ABI, true)
+
+export const useAllocation = (): CurrencyAmount<Token> => {
+  const { chainId, account } = useActiveWeb3React()
+  const [allocation, setAllocation] = useState(CurrencyAmount.fromRawAmount(COW, 0))
+
+  useEffect(() => {
+    let canceled = false
+    if (account && chainId) {
+      fetchClaim(account, chainId).then((claim) => {
+        if (!canceled) {
+          setAllocation(CurrencyAmount.fromRawAmount(COW, claim?.amount ?? 0))
+        }
+      })
+    }
+    return () => {
+      canceled = true
+    }
+  }, [chainId, account])
+
+  return allocation
+}
+
+const START_TIME = 1644584715000
+const DURATION = 126144000000
+
+export const useBalances = () => {
+  const { account } = useActiveWeb3React()
+  const allocated = useAllocation()
+  const vested = allocated.multiply(Math.min(Date.now() - START_TIME, DURATION)).divide(DURATION)
+
+  const tokenDistro = useTokenDistroContract()
+  const { result, loading } = useSingleCallResult(allocated.greaterThan(0) ? tokenDistro : null, 'balances', [
+    account || undefined,
+  ])
+  const claimed = useMemo(() => CurrencyAmount.fromRawAmount(COW, result ? result.claimed.toString() : 0), [result])
+
+  return {
+    allocated,
+    vested,
+    claimed,
+    loading,
+  }
+}
+
+interface ClaimCallbackParams {
+  openModal: (message: string, operationType: OperationType) => void
+  closeModal: () => void
+  isFirstClaim: boolean
+}
+export function useClaimCallback({
```

Comment:
Same for this. I suggest having a name that gives more context on what's claim is this to not be confused with the airdrop claiming